src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -27,9 +27,9 @@ private ReadFilterLibrary(){ /*no instance*/ }
     public static final ReadFilter NON_ZERO_REFERENCE_LENGTH_ALIGNMENT = read -> read.getCigarElements().stream().anyMatch(c -> c.getOperator().consumesReferenceBases() && c.getLength() > 0);
 
     /**
-     * Reads whose mate maps to the same contig.
+     * Reads that either have a mate that maps to the same contig, or don't have a mapped mate.
      */
-    public static final ReadFilter MATE_ON_SAME_CONTIG = read -> read.isPaired() && ! read.isUnmapped() && ! read.mateIsUnmapped() && read.getContig().equals(read.getMateContig());
+    public static final ReadFilter MATE_ON_SAME_CONTIG = read -> ! read.isPaired() || read.isUnmapped() || read.mateIsUnmapped() || read.getContig().equals(read.getMateContig());
```

Comment:
This change was to make our `MATE_ON_SAME_CONTIG` filter consistent with GATK3's `BadMateFilter` (albeit with the logic inverted), since the `HaplotypeCaller` needs this filter. Will try to come up with a more accurate name.